### PR TITLE
Sagemaker add pipeline tags

### PR DESCRIPTION
**Pull request by htahir1**
## Describe changes
I added tags to sagemaker pipelines

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by htahir1** [comment]
@bcdurak i have not tested this, so if u can test it id be grateful

File: src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py [hunk 1]
```diff
@@ -332,6 +332,18 @@ def prepare_or_run_pipeline(
                     else None
                 ),
             )
+            pipeline_tags = step_settings.pipeline_tags
+            args_for_step_executor.setdefault(
```
Comment (schustmi) [inline src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py:336]: Why are we passing these here as well? I couldn't find an argument `pipeline_tags` for the estimator class that this gets passed to.

Comment (htahir1) [reply src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py:336]: oh wait i thought it was doing something completely different, but I removed this now

File: src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py [hunk 2]
```diff
@@ -332,6 +332,7 @@ def prepare_or_run_pipeline(
                     else None
                 ),
             )
+            pipeline_tags = step_settings.pipeline_tags
```
Comment (schustmi) [inline src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py:335]: Ah another thing: This is taking the pipeline tags from the settings specified for a specific step? I'm guessing this is in a for-loop and will just use the settings of the latest step.

Instead, I think we should use the tags defined on the pipeline settings (`self.get_settings(deployment)`)

**Review by schustmi — CHANGES_REQUESTED** [review]

File: src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py (hunk 2 above)
Comment (htahir1) [reply src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py:335]: pfft right again. Im rusty AF. Sorry

**Review by schustmi — APPROVED** [review]
🦭

File: docs/book/component-guide/orchestrators/sagemaker.md [hunk 3]
```diff
@@ -290,6 +290,53 @@ sagemaker_orchestrator_settings = SagemakerOrchestratorSettings(
 Using multichannel output or output mode except `EndOfJob` will make it impossible to use TrainingStep and also Warm Pools. See corresponding section of this document for details.
 {% endhint %}
 
+You're absolutely right. Showing examples of how to apply these settings at both the @pipeline and @step level would be much more helpful and illustrative for users. Let's revise the code snippets to demonstrate this:
```
Comment (htahir1) [inline docs/book/component-guide/orchestrators/sagemaker.md:293]:
```suggestion
```

File: docs/book/component-guide/orchestrators/sagemaker.md [hunk 4]
```diff
@@ -290,6 +290,53 @@ sagemaker_orchestrator_settings = SagemakerOrchestratorSettings(
...
+    preprocess_data()
+    # Other steps...
+
+# Run the pipeline
+my_training_pipeline()
+```
+
+In this example:
+
+- The `pipeline_tags` are applied to the entire SageMaker pipeline execution.
```
Comment (htahir1) [inline docs/book/component-guide/orchestrators/sagemaker.md:335]:
```suggestion
- The `pipeline_tags` are applied to the entire SageMaker pipeline execution. They are also propagated automatically by AWS down to the child Sagemaker jobs.
```

File: docs/book/component-guide/orchestrators/sagemaker.md [hunk 5]
```diff
@@ -290,6 +290,52 @@ sagemaker_orchestrator_settings = SagemakerOrchestratorSettings(
 Using multichannel output or output mode except `EndOfJob` will make it impossible to use TrainingStep and also Warm Pools. See corresponding section of this document for details.
 {% endhint %}
 
+
```
Comment (htahir1) [inline docs/book/component-guide/orchestrators/sagemaker.md:293]:
```suggestion
```

File: docs/book/component-guide/orchestrators/sagemaker.md [hunk 6]
```diff
@@ -290,6 +290,51 @@ sagemaker_orchestrator_settings = SagemakerOrchestratorSettings(
...
+    preprocess_data()
+    # Other steps...
+
+# Run the pipeline
+my_training_pipeline()
+```
+
+In this example:
+
+- The `pipeline_tags` are applied to the entire SageMaker pipeline execution. They are also propagated automatically by AWS down to the child Sagemaker jobs.
```
Comment (htahir1) [inline docs/book/component-guide/orchestrators/sagemaker.md:333]:
```suggestion
- The `pipeline_tags` are applied to the entire SageMaker pipeline object. SageMaker automatically applies the pipeline_tags to all its associated jobs. 
```

**Review by bcdurak — APPROVED** [review]
LGTM. Tested this on our AWS stack, everything is working as intended.

**Pull request merged**